### PR TITLE
Remove IVF_FLAT_NM

### PR DESCRIPTION
**Pull request by cydrain**
Issue: #30

### CODE REVIEW

**Comment by cydrain** [comment]
build Milvus dev image:
https://jenkins.milvus.io:18080/blue/organizations/jenkins/milvus-images%2Fbuild-dev-image/detail/build-dev-image/944/pipeline/

File: src/index/ivf/ivf.cc
Comment (cqy123456) [inline src/index/ivf/ivf.cc:1]: Should IndexIVFFlat return true in  HasRawData function() ?

**Comment by cydrain** [comment]
https://jenkins.milvus.io:18080/blue/organizations/jenkins/milvus-images%2Fbuild-dev-image/detail/build-dev-image/951/pipeline

File: src/common/utils.cc [hunk 1]
```diff
@@ -14,13 +14,19 @@
...
+#include "faiss/impl/FaissException.h"
+#include "faiss/index_io.h"
+#include "io/FaissIO.h"
 #include "knowhere/log.h"
 #include "simd/hook.h"
 
 namespace knowhere {
 
 const float FloatAccuracy = 0.00001;
+const char* IVF_FLAT_NM_MAGIC = "IVF_FLAT_NM";
```
Comment (cqy123456) [inline src/common/utils.cc:29]: This variable has no place to use, we can delete it.

**Comment by cydrain** [comment]
Milvus image:
https://jenkins.milvus.io:18080/job/milvus-images/job/build-dev-image/960/
harbor.milvus.io/milvus/milvus:cydrain-caiyd_remove_ivf_flat_nm-cae4b42-20230905

Upgrade test:
https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_cron/detail/deploy_test_cron/1310/pipeline

**Comment by cydrain** [comment]
Milvus image:
harbor.milvus.io/milvus/milvus:cydrain-caiyd_remove_ivf_flat_nm-8177c05-20230907

Upgrade test:

**Pull request merged**